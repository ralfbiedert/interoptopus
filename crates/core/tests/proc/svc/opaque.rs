use interoptopus::{ffi, ffi_service, ffi_type};

#[ffi_type]
enum Error {
    Something,
}

#[ffi_type(opaque)]
struct Service;

#[ffi_service]
impl Service {
    pub fn new() -> ffi::Result<Self, Error> {
        ffi::Ok(Self)
    }
}

fn main() {}
