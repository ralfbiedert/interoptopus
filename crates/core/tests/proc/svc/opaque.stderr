error[E0080]: evaluation of constant value failed
  --> tests/proc/svc/opaque.rs:13:12
   |
13 |     pub fn new() -> ffi::Result<Self, Error> {
   |            ^^^ evaluation panicked: This method looks like a constructor, but does not return ffi::Result<Self, _>
